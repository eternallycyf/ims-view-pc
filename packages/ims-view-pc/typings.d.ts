import '@umijs/max/typings';

declare module '*.less';

declare module 'react' {
  interface CSSProperties {
    [key: `--${string}`]: string;
  }
}

import { SlateDescendant, SlateElement, SlateText } from '@wangeditor/editor/dist/';

declare module '@wangeditor/editor' {
  // 扩展 Text
  interface SlateText {
    text: string;
  }

  // 扩展 Element
  interface SlateElement {
    type: string;
    children: SlateDescendant[];
  }

  type VideoElement = SlateElement & {
    src: string;
    poster?: string;
  };
}
