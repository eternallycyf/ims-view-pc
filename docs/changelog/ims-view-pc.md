---
title: ims-view-pc
order:
nav:
  title: 更新日志
---

<embed src="../../packages/ims-view-pc/CHANGELOG.md"></embed>
