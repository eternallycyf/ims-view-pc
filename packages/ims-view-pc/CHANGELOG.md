# Changelog

### [Version&nbsp;1.22.6](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.5...ims-view-pc@1.22.6)

<sup>Released on **2023-11-25**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.10.0

### [Version&nbsp;1.22.5](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.4...ims-view-pc@1.22.5)

<sup>Released on **2023-11-25**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.9.0

### [Version&nbsp;1.22.4](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.3...ims-view-pc@1.22.4)

<sup>Released on **2023-11-25**</sup>

#### üìù ÊñáÊ°£

- **ims-view-pc**: Update docs.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### Documentation

- **ims-view-pc**: Update docs ([adaedd9](https://github.com/eternallycyf/ims-view-pc/commit/adaedd9))

</details>

### Dependencies

- **@ims-view/chart:** upgraded to 1.3.0

### [Version&nbsp;1.22.3](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.2...ims-view-pc@1.22.3)

<sup>Released on **2023-11-24**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/chart:** upgraded to 1.2.0

### [Version&nbsp;1.22.2](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.1...ims-view-pc@1.22.2)

<sup>Released on **2023-11-24**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.8.0

### [Version&nbsp;1.22.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.22.0...ims-view-pc@1.22.1)

<sup>Released on **2023-11-23**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/hooks:** upgraded to 1.10.0

## [Version&nbsp;1.22.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.21.0...ims-view-pc@1.22.0)

<sup>Released on **2023-11-21**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add WaterMark and update docs.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add WaterMark and update docs ([5d4be30](https://github.com/eternallycyf/ims-view-pc/commit/5d4be30))

</details>

## [Version&nbsp;1.21.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.20.1...ims-view-pc@1.21.0)

<sup>Released on **2023-11-21**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Update docs feature.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Update docs feature ([eeca388](https://github.com/eternallycyf/ims-view-pc/commit/eeca388))

</details>

### [Version&nbsp;1.20.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.20.0...ims-view-pc@1.20.1)

<sup>Released on **2023-11-20**</sup>

#### üêõ ‰øÆÂ§ç

- FormatColumns title can use function and string.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- FormatColumns title can use function and string ([cff7281](https://github.com/eternallycyf/ims-view-pc/commit/cff7281))

</details>

## [Version&nbsp;1.20.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.19.1...ims-view-pc@1.20.0)

<sup>Released on **2023-11-18**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add muitiple formList and commonEditTable demo.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add muitiple formList and commonEditTable demo ([78eb980](https://github.com/eternallycyf/ims-view-pc/commit/78eb980))

</details>

### [Version&nbsp;1.19.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.19.0...ims-view-pc@1.19.1)

<sup>Released on **2023-11-18**</sup>

#### üêõ ‰øÆÂ§ç

- **commonEditTable**: Update formItem types in editTable.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **commonEditTable**: Update formItem types in editTable ([b304766](https://github.com/eternallycyf/ims-view-pc/commit/b304766))

</details>

## [Version&nbsp;1.19.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.18.0...ims-view-pc@1.19.0)

<sup>Released on **2023-11-18**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add feature component with CustomMenu CustomTab.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add feature component with CustomMenu CustomTab ([8fe34f5](https://github.com/eternallycyf/ims-view-pc/commit/8fe34f5))

</details>

## [Version&nbsp;1.18.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.17.1...ims-view-pc@1.18.0)

<sup>Released on **2023-11-18**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add custom theme docs.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add custom theme docs ([3463a1e](https://github.com/eternallycyf/ims-view-pc/commit/3463a1e))

</details>

### [Version&nbsp;1.17.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.17.0...ims-view-pc@1.17.1)

<sup>Released on **2023-11-15**</sup>

#### üêõ ‰øÆÂ§ç

- RenderFormItem col span use ?? replace ||.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- RenderFormItem col span use ?? replace || ([01639f1](https://github.com/eternallycyf/ims-view-pc/commit/01639f1))

</details>

## [Version&nbsp;1.17.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.16.0...ims-view-pc@1.17.0)

<sup>Released on **2023-11-05**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add CustomForm.update.

#### üìù ÊñáÊ°£

- Add CommonEditTable apis, update footer link and commonEditTable apiHeader.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add CustomForm.update ([db4ef4b](https://github.com/eternallycyf/ims-view-pc/commit/db4ef4b))

#### Documentation

- Add CommonEditTable apis ([45999a1](https://github.com/eternallycyf/ims-view-pc/commit/45999a1))
- Update footer link and commonEditTable apiHeader ([d496d7c](https://github.com/eternallycyf/ims-view-pc/commit/d496d7c))

</details>

## [Version&nbsp;1.16.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.15.0...ims-view-pc@1.16.0)

<sup>Released on **2023-10-31**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: CommonEditTable add custom edit row.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: CommonEditTable add custom edit row ([079d124](https://github.com/eternallycyf/ims-view-pc/commit/079d124))

</details>

## [Version&nbsp;1.15.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.14.0...ims-view-pc@1.15.0)

<sup>Released on **2023-10-30**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add endregion.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add endregion ([8050064](https://github.com/eternallycyf/ims-view-pc/commit/8050064))

</details>

## [Version&nbsp;1.14.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.13.1...ims-view-pc@1.14.0)

<sup>Released on **2023-10-29**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add CommonEditTable.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add CommonEditTable ([81e9f63](https://github.com/eternallycyf/ims-view-pc/commit/81e9f63))

</details>

### [Version&nbsp;1.13.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.13.0...ims-view-pc@1.13.1)

<sup>Released on **2023-10-28**</sup>

#### üé´ ÊùÇÈ°π

- : remove "type": "module",.

#### üêõ ‰øÆÂ§ç

- RenderFormItem use lazy replace required.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### Chores

- : remove "type": "module", ([aafce5b](https://github.com/eternallycyf/ims-view-pc/commit/aafce5b))

#### What's fixed

- RenderFormItem use lazy replace required ([cc054ae](https://github.com/eternallycyf/ims-view-pc/commit/cc054ae))

</details>

## [Version&nbsp;1.13.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.12.0...ims-view-pc@1.13.0)

<sup>Released on **2023-10-27**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Add Editor formItem and adjust import path.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add Editor formItem and adjust import path ([f9608a2](https://github.com/eternallycyf/ims-view-pc/commit/f9608a2))

</details>

## [Version&nbsp;1.12.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.11.0...ims-view-pc@1.12.0)

<sup>Released on **2023-10-27**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add ErrorBoundary.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add ErrorBoundary ([6727981](https://github.com/eternallycyf/ims-view-pc/commit/6727981))

</details>

## [Version&nbsp;1.11.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.10.1...ims-view-pc@1.11.0)

<sup>Released on **2023-10-27**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add Ellipsis.

#### üé´ ÊùÇÈ°π

- : add runes2 and resize-observer-polyfill.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add Ellipsis ([5cf8a08](https://github.com/eternallycyf/ims-view-pc/commit/5cf8a08))

#### Chores

- : add runes2 and resize-observer-polyfill ([ade537e](https://github.com/eternallycyf/ims-view-pc/commit/ade537e))

</details>

### [Version&nbsp;1.10.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.10.0...ims-view-pc@1.10.1)

<sup>Released on **2023-10-27**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/hooks:** upgraded to 1.9.0

## [Version&nbsp;1.10.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.9.0...ims-view-pc@1.10.0)

<sup>Released on **2023-10-15**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add editor.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add editor ([ae2579d](https://github.com/eternallycyf/ims-view-pc/commit/ae2579d))

</details>

## [Version&nbsp;1.9.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.8.0...ims-view-pc@1.9.0)

<sup>Released on **2023-10-14**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add CustForm.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add CustForm ([8ba413a](https://github.com/eternallycyf/ims-view-pc/commit/8ba413a))

</details>

## [Version&nbsp;1.8.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.7.0...ims-view-pc@1.8.0)

<sup>Released on **2023-10-13**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add CommonCard.CardList.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add CommonCard.CardList ([25bb2b7](https://github.com/eternallycyf/ims-view-pc/commit/25bb2b7))

</details>

### Dependencies

- **@ims-view/hooks:** upgraded to 1.8.0

## [Version&nbsp;1.7.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.6.0...ims-view-pc@1.7.0)

<sup>Released on **2023-10-12**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **packages**: Update üìù docs:.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **packages**: Update üìù docs: ([5a75b79](https://github.com/eternallycyf/ims-view-pc/commit/5a75b79))

</details>

### Dependencies

- **@ims-view/hooks:** upgraded to 1.7.0
- **@ims-view/utils:** upgraded to 1.7.0

## [Version&nbsp;1.6.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.5.3...ims-view-pc@1.6.0)

<sup>Released on **2023-10-11**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **@ims-view/chart**: üéâ init chart packages.

#### üé´ ÊùÇÈ°π

- Antd use last version.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/chart**: üéâ init chart packages ([5f60104](https://github.com/eternallycyf/ims-view-pc/commit/5f60104))

#### Chores

- Antd use last version ([c2f7e13](https://github.com/eternallycyf/ims-view-pc/commit/c2f7e13))

</details>

### [Version&nbsp;1.5.3](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.5.2...ims-view-pc@1.5.3)

<sup>Released on **2023-10-11**</sup>

#### üêõ ‰øÆÂ§ç

- **ims-view-pc**: VideoViewer css import by tsx.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **ims-view-pc**: VideoViewer css import by tsx ([11ba7b7](https://github.com/eternallycyf/ims-view-pc/commit/11ba7b7))

</details>

### [Version&nbsp;1.5.2](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.5.1...ims-view-pc@1.5.2)

<sup>Released on **2023-10-11**</sup>

#### üêõ ‰øÆÂ§ç

- **ims-view-pc**: VideoViewer css use unpkg.com.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **ims-view-pc**: VideoViewer css use unpkg.com ([58f0441](https://github.com/eternallycyf/ims-view-pc/commit/58f0441))

</details>

### [Version&nbsp;1.5.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.5.0...ims-view-pc@1.5.1)

<sup>Released on **2023-10-10**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

## [Version&nbsp;1.5.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.4.0...ims-view-pc@1.5.0)

<sup>Released on **2023-10-10**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Ims-view:CommonCard.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Ims-view:CommonCard ([0132a1b](https://github.com/eternallycyf/ims-view-pc/commit/0132a1b))

</details>

## [Version&nbsp;1.4.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.3.1...ims-view-pc@1.4.0)

<sup>Released on **2023-10-09**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Ims-view:AccessBtn.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Ims-view:AccessBtn ([fbcc38b](https://github.com/eternallycyf/ims-view-pc/commit/fbcc38b))

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.6.0

### [Version&nbsp;1.3.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.3.0...ims-view-pc@1.3.1)

<sup>Released on **2023-10-08**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

## [Version&nbsp;1.3.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.2.0...ims-view-pc@1.3.0)

<sup>Released on **2023-10-07**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Ims-view:VidemoViewer.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Ims-view:VidemoViewer ([3c434e5](https://github.com/eternallycyf/ims-view-pc/commit/3c434e5))

</details>

### Dependencies

- **@ims-view/hooks:** upgraded to 1.6.0

## [Version&nbsp;1.2.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.1.1...ims-view-pc@1.2.0)

<sup>Released on **2023-10-04**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Ims-view:AudioPlayer + @ims-view/hooks: refactor.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Ims-view:AudioPlayer + @ims-view/hooks: refactor ([7825a35](https://github.com/eternallycyf/ims-view-pc/commit/7825a35))

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.5.0
- **@ims-view/hooks:** upgraded to 1.5.0

### [Version&nbsp;1.1.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.1.0...ims-view-pc@1.1.1)

<sup>Released on **2023-10-04**</sup>

#### üêõ ‰øÆÂ§ç

- Demo use fetch replace axios.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- Demo use fetch replace axios ([296780b](https://github.com/eternallycyf/ims-view-pc/commit/296780b))

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.4.1
- **@ims-view/hooks:** upgraded to 1.4.1

## [Version&nbsp;1.1.0](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.0.1...ims-view-pc@1.1.0)

<sup>Released on **2023-10-03**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- Ims-view:CommonDemo + @ims-view/hooks:useFetch.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Ims-view:CommonDemo + @ims-view/hooks:useFetch ([2107502](https://github.com/eternallycyf/ims-view-pc/commit/2107502))

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.4.0
- **@ims-view/hooks:** upgraded to 1.4.0

### [Version&nbsp;1.0.1](https://github.com/eternallycyf/ims-view-pc/compare/ims-view-pc@1.0.0...ims-view-pc@1.0.1)

<sup>Released on **2023-10-03**</sup>

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

</details>

### Dependencies

- **@ims-view/utils:** upgraded to 1.3.1

## Version&nbsp;1.0.0

<sup>Released on **2023-10-02**</sup>

#### ‚ú® Êñ∞ÁâπÊÄß

- **ims-view-pc**: Add ims-view packages.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **ims-view-pc**: Add ims-view packages ([8e71a2a](https://github.com/eternallycyf/ims-view-pc/commit/8e71a2a))

</details>

<div align="right">

[![](https://img.shields.io/badge/-BACK_TO_TOP-151515?style=flat-square)](#readme-top)

</div>

### Dependencies

- **@ims-view/utils:** upgraded to 1.3.0
- **@ims-view/hooks:** upgraded to 1.3.0
