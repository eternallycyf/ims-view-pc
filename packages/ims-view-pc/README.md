# ims-view-pc

| 组件                                                             | 下载量                                                                                                       | 版本                                                                             |
| ---------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------- |
| [ims-view-pc](https://www.npmjs.com/package/ims-view-pc)         | ![npm-image](http://img.shields.io/npm/v/ims-view-pc.svg?style=flat-square&color=deepgreen&label=latest)     | [![NPM downloads][ims-view-pc-download-image]][ims-view-pc-download-url]         |
| [@ims-view/chart](https://www.npmjs.com/package/@ims-view/chart) | ![npm-image](http://img.shields.io/npm/v/@ims-view/chart?style=flat-square&color=deepgreen&label=latest)     | [![NPM downloads][@ims-view/chart-download-image]][@ims-view/chart-download-url] |
| [@ims-view/hooks](https://www.npmjs.com/package/@ims-view/hooks) | ![npm-image](http://img.shields.io/npm/v/@ims-view/hooks.svg?style=flat-square&color=deepgreen&label=latest) | [![NPM downloads][@ims-view/hooks-download-image]][@ims-view/hooks-download-url] |
| [@ims-view/utils](https://www.npmjs.com/package/@ims-view/utils) | ![npm-image](http://img.shields.io/npm/v/@ims-view/utils.svg?style=flat-square&color=deepgreen&label=latest) | [![NPM downloads][@ims-view/utils-download-image]][@ims-view/utils-download-url] |

[ims-view-pc-download-url]: https://npmjs.org/package/ims-view-pc
[ims-view-pc-download-image]: https://img.shields.io/npm/dm/ims-view-pc.svg?style=flat-square
[@ims-view/chart-download-url]: https://npmjs.org/package/@ims-view/chart
[@ims-view/chart-download-image]: https://img.shields.io/npm/dm/@ims-view/chart?style=flat-square
[@ims-view/hooks-download-url]: https://npmjs.org/package/@ims-view/hooks
[@ims-view/hooks-download-image]: https://img.shields.io/npm/dm/@ims-view/hooks.svg?style=flat-square
[@ims-view/utils-download-url]: https://npmjs.org/package/@ims-view/utils
[@ims-view/utils-download-image]: https://img.shields.io/npm/dm/@ims-view/utils.svg?style=flat-square

## License

[MIT](../../LICENSE) ® eternallycyf

<!-- docs url -->

- https://ims-view.site/components/

## required dev

```JSON
{
    "react": "^18",
    "react-dom": "^18",
    "@ant-design/icons": "^5.2.6",
    "antd": "^5",
    "lodash": "^4.17.21",
    "axios": "^1.5.1"
}
```
