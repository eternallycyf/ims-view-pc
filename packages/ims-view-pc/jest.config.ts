import { Config } from '@umijs/test';
import base from '../../jest.config.base';

const packageName = 'ims-view-pc';

const root = '<rootDir>/packages/ims-view-pc';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
};

export default config;
