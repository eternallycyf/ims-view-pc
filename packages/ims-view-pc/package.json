{
  "name": "ims-view-pc",
  "version": "1.22.5",
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib",
    "es",
    "dist"
  ],
  "scripts": {
    "build": "father build",
    "clean": "rm -rf es lib dist build coverage .eslintcache",
    "cov": "jest --coverage",
    "doctor": "father doctor",
    "prepublishOnly": "npm run doctor && npm run build",
    "start": "father dev",
    "test": "jest"
  },
  "dependencies": {
    "@babel/runtime": "^7.23.1",
    "@ims-view/chart": "1.3.0",
    "@ims-view/hooks": "1.10.0",
    "@ims-view/utils": "1.9.0",
    "classnames": "^2.3.2",
    "video.js": "^7.21.5",
    "dayjs": "^1.11.10",
    "react-use": "^17.4.0",
    "runes2": "^1.1.2",
    "resize-observer-polyfill": "1.5.1",
    "@wangeditor/editor": "^5.1.23",
    "@wangeditor/editor-for-react": "^1.0.6"
  },
  "peerDependencies": {
    "@ant-design/icons": "^5.2.6",
    "antd": "^5",
    "axios": "^1.5.1",
    "lodash": "^4.17.21",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@ant-design/icons": "^5.2.6",
    "@types/video.js": "^7.3.53",
    "antd": "^5",
    "axios": "^1.5.1",
    "lodash": "^4.17.21"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  }
}
