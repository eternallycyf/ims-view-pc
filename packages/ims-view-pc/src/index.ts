export { default as AccessBtn } from './components/AccessBtn';
export * from './components/AccessBtn/interface';
export { default as AudioPlayer } from './components/AudioPlayer';
export * from './components/AudioPlayer/interface';
export { default as CheckModal } from './components/CheckModal';
export * from './components/CheckModal/interface';
export { default as Clipboard } from './components/Clipboard';
export * from './components/Clipboard/interface';
export { default as CommonCard } from './components/CommonCard';
export * from './components/CommonCard/interface';
export { default as CommonDemo } from './components/CommonDemo';
export * from './components/CommonDemo/interface';
export { default as CommonDescriptions } from './components/CommonDescriptions';
export * from './components/CommonDescriptions/interface';
export { default as CommonEditTable } from './components/CommonEditTable';
export * from './components/CommonEditTable/interface';
export { default as CommonSearch } from './components/CommonSearch';
export * from './components/CommonSearch/interface';
export { default as CommonTable } from './components/CommonTable';
export * from './components/CommonTable/interface';
export type {
  ICascadeControlProps,
  IEditorProps,
  IMentionsControlProps,
  ISimpleControlProps,
  IUpdateControlProps,
} from './components/CustomForm/interface';
export { default as CustomModal } from './components/CustomModal';
export type { FormModalProps, ICustomModalFormList } from './components/CustomModal/form';
export { default as Ellipsis } from './components/Ellipsis';
export * from './components/Ellipsis/interface';
export { default as ErrorBoundary } from './components/ErrorBoundary';
export * from './components/ExpandSelect';
export { default as ExpandSelect } from './components/ExpandSelect';
export { default as ExportButton } from './components/ExportButton';
export * from './components/ExportButton/interface';
export { default as Flow } from './components/Flow';
export * from './components/Flow/interface';
export { default as Icon } from './components/Icon';
export { default as Marker } from './components/Marker';
export * from './components/Marker/interface';
export { default as MutateObserver } from './components/MutateObserver';
export * from './components/MutateObserver/interface';
export { default as Piano } from './components/Piano';
export * from './components/Piano/interface';
export { default as Portal } from './components/Portal';
export * from './components/Portal/interface';
export { default as SectionTitle } from './components/SectionTitle';
export * from './components/SectionTitle/interface';
export { default as Sticky } from './components/Sticky';
export * from './components/Sticky/interface';
export { default as Theme } from './components/theme';
export * from './components/theme/interface';
export { default as ToggleBtn } from './components/ToggleBtn';
export * from './components/ToggleBtn/interface';
export { default as TreeModal } from './components/TreeModal';
export * from './components/TreeModal/interface';
export { default as VideoViewer } from './components/VideoViewer';
export * from './components/VideoViewer/interface';
export { default as WaterMark } from './components/WaterMark';
export * from './components/WaterMark/interface';
export { getFieldComp, renderFormItem } from './core/helpers';
export { CustomTheme, variables } from './styles/variables';
export * from './type/index';
