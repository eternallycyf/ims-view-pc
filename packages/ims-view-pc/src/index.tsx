import React, { FC } from 'react';

const Bar: FC = () => {
  return <div>hello ims-view-pc!</div>;
};

export default Bar;
