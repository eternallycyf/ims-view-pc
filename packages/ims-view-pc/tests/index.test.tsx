import { render } from '@testing-library/react';
import axios from 'axios';
import { CommonDemo } from 'ims-view-pc';
import React from 'react';

test('CommonDemo', () => {
  const props = {
    fetchConfig: {
      apiUrl: '/fetchUserInfo',
      method: 'post' as 'post',
      dataPath: 'data',
    },
    request: axios,
    initRequest: false,
    dataHandler: (data) => data?.data,
  };
  const { container } = render(<CommonDemo {...props} />);

  expect(container).toMatchSnapshot();
});
