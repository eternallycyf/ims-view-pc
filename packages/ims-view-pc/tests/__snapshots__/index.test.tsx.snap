// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils 1`] = `
<div>
  <div>
    hello ims-view-pc!
  </div>
</div>
`;
